packages/roosterjs-editor-core/lib/editor/Editor.ts

```diff
@@ -115,6 +117,9 @@ export default class Editor implements IEditor {
             trustedHTMLHandler: options.trustedHTMLHandler || ((html: string) => html),
             zoomScale: zoomScale,
             sizeTransformer: options.sizeTransformer || ((size: number) => size / zoomScale),
+            getVisibleViewport:
+                options.getVisibleViewport ||
+                (() => normalizeRect(this.getScrollContainer().getBoundingClientRect())!),

```

Comment:
If scroll container is not the editor DIV, we need to calculate both size and get the intersected area.